include/grpc/grpc_security_constants.h

```diff
@@ -105,6 +106,28 @@ typedef enum {
   GRPC_SSL_REQUEST_AND_REQUIRE_CLIENT_CERTIFICATE_AND_VERIFY
 } grpc_ssl_client_certificate_request_type;
 
+
+typedef enum {
+  /** Default option: performs server certificate verification and hostname
+     verification */
+  GRPC_SSL_SERVER_VERIFICATION,
+  /** Performs server certificate verification, but skips hostname verification
+   */
+  GRPC_SSL_SKIP_HOSTNAME_VERIFICATION,
+  /** Performs hostname name verification, but skips server certificate
+     verification */
+  GRPC_SSL_SKIP_SERVER_CERTIFICATE_VERIFICATION,
```

Comment:
It does not make sense to do hostname verification check without validating server's certificate since the the information used in hostname verification check also comes from server's certificate.

Reply:
This is as per @jiangtaoli2016 suggestion. @jiangtaoli2016 ?

Reply:
I have synced with @yihuazhang. Let's do three enums: default verification, skip hostname verification, skip both hostname verification and server cert verification.

Reply:
Updated the PR and removed `GRPC_SSL_SKIP_SERVER_CERTIFICATE_VERIFICATION`.